webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactStorage.scala

```diff
@@ -696,7 +708,7 @@ class LdapNodeFactStorage(
                           _ <- ZIO.foreach(newSoft)(x => con.save(inventoryMapper.entryFromSoftware(x)))
                         } yield Some(newSoft.map(_.id) ++ merged.alreadySavedSoftware.map(_.id))
                     }
-      optOld     <- getNodeFact(nodeFact.id, nodeFact.rudderSettings.status, SelectFacts.noSoftware).map(
+      optOld     <- getNodeFact(nodeFact.id, nodeFact.rudderSettings.status, attrs).map(
```

Comment:
here, we want to know the really reallity. If we asked for software, we need them. 
But in the opposite, is we didn't asked for inventory parts, we don't need them. 
(if we need one, then we need all, since inventory save delete missing bits)